src/components/MainNav/MainNav.js

```diff
@@ -86,6 +99,39 @@ class MainNav extends Component {
     this.context.router.history.push('/');
   }
 
+  subscribeToQueryChanges(moduleInfo) {
+    if (this.unsub) this.unsub();
+    if (!moduleInfo.queryResource) return;
+    this.unsub = this.store.subscribe(() => {
+      const previousQueryValues = this.queryValues;
+      // This is not DRY, as it was expressed already in LocalResource is stripes-connect,
+      // And in Root.js in stripes-core. Both State Keys should be derived from a common mechanism.
+      this.queryValues = this.store.getState()[`${moduleInfo.dataKey ? `${moduleInfo.dataKey}#` : ''}${_.snakeCase(moduleInfo.module)}_${moduleInfo.queryResource}`];
+      if (previousQueryValues !== this.queryValues) {
+        // This is not DRY, as it was expressed already in utils/transitionToParams is stripes-connect,
+        // It is changed slightly, but I did not want to make changes to a method that is being used elswhere.
+        const location = this.props.location;
+        const query = location.query ? location.query : queryString.parse(location.search);
+        const allParams = Object.assign({}, query, this.queryValues);
+        const keys = Object.keys(allParams);
+
+        let url = location.pathname;
+
+        if (keys.length) {
+          url += `?${queryString.stringify(allParams)}`;
+        }
+        this.props.history.replace(url);
```

Comment:
Why was `replace` the right choice here rather than `push` as usual?

Reply:
In this instance we want the shape of the URL to change, but not trigger any navigation.